src/sw.ts

```diff
@@ -407,8 +423,29 @@ async function fetchHandler ({ path, request }: FetchHandlerArg): Promise<Respon
    * * https://bugs.chromium.org/p/chromium/issues/detail?id=823697
    * * https://bugzilla.mozilla.org/show_bug.cgi?id=1394102
    */
-  // 5 minute timeout
-  const signal = AbortSignal.timeout(5 * 60 * 1000)
+  const abortController = new AbortController()
+  const signal = abortController.signal
+  const abortFn = (event: Pick<AbortSignalEventMap['abort'], 'type'>): void => {
+    clearTimeout(signalAbortTimeout)
+    if (event?.type !== 'verified-fetch-timeout') {
+      log.trace('helia-sw: request signal aborted')
+      abortController.abort('request signal aborted')
+    } else {
+      log.trace('helia-sw: timeout waiting for response from @helia/verified-fetch')
+      abortController.abort('timeout')
+    }
+  }
+  /**
+   * five minute delay to get the initial response.
+   *
+   * @todo reduce to 2 minutes?
+   */
+  const signalAbortTimeout = setTimeout(() => {
+    abortFn({ type: 'verified-fetch-timeout' })
+  }, 5 * 60 * 1000)
+  // if the fetch event is aborted, we need to abort the signal we give to @helia/verified-fetch
+  event.request.signal.addEventListener('abort', abortFn)
```

Comment:
We might want to remove this eventListener after the verifiedFetch promise is resolved/rejected to avoid memory leaks.

Reply:
IIUC correctly the `event.request` is dead (and all things attached to it) as soon as the request is consumed, but that's a good CYA improvement

Reply:
updated.

Reply:
Actually, we can't remove the event listener for this in a way that makes sense, because the browser controls the request and we would need to remove the listener after consumption of the response body.